src/packageurl/contrib/purl2url.py

```diff
@@ -0,0 +1,150 @@
+# download.
+
+from packageurl import PackageURL
+
+
+def build_bitbucket(**data):
+    """
+    Return a Bitbucket URL `url` from a mapping of Package URL data `data` 
+    """
+
+    name = data.get('name')
+    namespace = data.get('namespace')
+    if not (name and namespace):
+        return
+
+    url = 'https://bitbucket.org/{namespace}/{name}'.format(
+        namespace=namespace, name=name
+    )
+    version = data.get('version')
+    if version:
+        url = '{url}/src/{version}'.format(url=url, version=version)
+
+    subpath = data.get('subpath')
+    if subpath:
+        url = '{url}/{subpath}'.format(url=url, subpath=subpath)
+
+    return url
+
+
+def build_cargo(**data):
+    """
+    Return a Cargo URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    version = data.get('version')
+    if not (name and version):
+        return
+
+    return 'https://crates.io/api/v1/crates/{name}/{version}/download'.format(
+        name=name, version=version
+    )
+
+
+def build_github(**data):
+    """
+    Return a GitHub URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    namespace = data.get('namespace')
+    if not (name and namespace):
+        return
+
+    url = 'https://github.com/{namespace}/{name}'.format(namespace=namespace, name=name)
+
+    version = data.get('version')
+    if version:
+        url = '{url}/tree/{version}'.format(url=url, version=version)
+
+    subpath = data.get('subpath')
+    if subpath:
+        url = '{url}/{subpath}'.format(url=url, subpath=subpath)
+
+    return url
+
+
+def build_gitlab(**data):
+    """
+    Return a GitLab URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    namespace = data.get('namespace')
+
+    if not (name and namespace):
+        return
+
+    url = 'https://gitlab.com/{namespace}/{name}'.format(namespace=namespace, name=name)
+
+    version = data.get('version')
+    if version:
+        url = '{url}/-/tree/{version}'.format(url=url, version=version)
+
+    subpath = data.get('subpath')
+    if subpath:
+        url = '{url}/{subpath}'.format(url=url, subpath=subpath)
+
+    return url
+
+
+def build_gem(**data):
+    """
+    Return a RubyGem URL `url` from a mapping of Package URL data `data`
+    """
+    name = data.get('name')
+    version = data.get('version')
+
+    if not (name and version):
+        return
+
+    return 'https://rubygems.org/gems/{name}-{version}.gem'.format(
+        name=name, version=version
+    )
+
+
+def purl2url(purl):
+    """
+    Return a valid URL `url` according to the type of given PURL `purl`
+    """
+    purl_data = PackageURL.from_string(purl)
+    
+    converter = {
+        'bitbucket': build_bitbucket,
+        'cargo': build_cargo,
+        'gem': build_gem,
+        'github': build_github,
+        'gitlab': build_gitlab,
+    }
+
+    type = purl_data.type
+    if type in converter:
```

Comment:
Can we change this to 
```
if not type in converter:
    return
** rest of code**
```

This way we do not have an extra line of indentation.